aptos-move/aptos-release-builder/src/components/consensus_config.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::utils::*;
+use anyhow::Result;
+use aptos_types::on_chain_config::OnChainConsensusConfig;
+use move_model::{code_writer::CodeWriter, emit, emitln, model::Loc};
+
+pub fn generate_consensus_upgrade_proposal(
+    consensus_config: &OnChainConsensusConfig,
+    is_testnet: bool,
+) -> Result<Vec<(String, String)>> {
+    let mut result = vec![];
+
+    let gas_schedule_blob = bcs::to_bytes(consensus_config).unwrap();
+
+    assert!(gas_schedule_blob.len() < 65536);
+
+    let writer = CodeWriter::new(Loc::default());
+
+    emitln!(writer, "// Consensus config upgrade proposal\n");
+
+    if is_testnet {
```

Comment:
nit: This is getting a bit redundant. Refactor this so each component doesn't have to do this themselves?

Reply:
Yea I think I can refactor a bit here.